File: 백준/Bronze/2920. 음계/음계.cc
#include <iostream>
using namespace std;
bool acs(int *arr){
    for(int i = 1; i< 8; i++){
        if(arr[i] != arr[i-1]+1){
            return false;
        }
    }
    return true;
}
bool dec(int *arr){
    for(int i = 1; i< 8; i++){
        if(arr[i] != arr[i-1]-1) return false;
    }
    return true;
}

int main(){
    int arr[10] = {};
    for(int i = 0; i< 8; i++){
        cin >> arr[i];
    }
    if(acs(arr)) cout << "ascending\n";
    else{
        if(dec(arr)) cout <<"descending\n";
        else cout <<"mixed\n";
    }

}

File: 백준/Bronze/2920. 음계/음계.py
def acs(arr):
    for i in range(1,8):
        if arr[i] != arr[i-1]+1:
            return False
    return True

def des(arr):
    for i in range(1, 8):
        if arr[i] != arr[i-1]-1:
            return False
    return True

if __name__=='__main__':
    arr = list(map(int,input().split()))
    if acs(arr):
        print("ascending")
    else:
        if des(arr):
            print("descending")
        else:
            print("mixed")